### PR TITLE
Added scenario tests and changed help

### PR DESCRIPTION
**Pull request by williexu**
Fixes https://github.com/Azure/azure-cli/issues/3476
---
Added scenario tests
Also Fixed some help examples for `lock create` and `lock update`

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [na ] The PR has modified HISTORY.rst describing any customer-facing, functional changes. Note that this does not include changes only to help content. (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

### Command Guidelines

- [na ] Each command and parameter has a meaningful description.
- [na ] Each new command has a test.

(see [Authoring Command Modules](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules))

### CODE REVIEW

File: src/command_modules/azure-cli-resource/azure/cli/command_modules/resource/tests/test_locks.py [hunk 1]
```diff
@@ -29,31 +29,60 @@ def test_lock_create_list_delete(self):
             self.cmd('az lock delete -n {}'.format(lock_name))
             self._sleep_for_lock_operation()
 
-    def test_readonly_lock_create_list_delete_resource_group(self):
-        self._lock_operation_with_resource_group('ReadOnly')
+    @ResourceGroupPreparer(name_prefix='cli_test_readonly_lock_create_list_delete_resource_group')
+    def test_readonly_lock_create_list_delete_resource_group(self, resource_group):
```
Comment (tjprescott) [inline src/command_modules/azure-cli-resource/azure/cli/command_modules/resource/tests/test_locks.py]: Recommend rename: "test_readonly_resource_group_lock" for the method and prefix.

File: src/command_modules/azure-cli-resource/azure/cli/command_modules/resource/tests/test_locks.py [hunk 2]
```diff
@@ -29,31 +29,60 @@ def test_lock_create_list_delete(self):
...
+        self._lock_operation_with_resource_group('ReadOnly', resource_group)
 
-    def test_cannotdelete_lock_create_list_delete_resource_group(self):
-        self._lock_operation_with_resource_group('CanNotDelete')
+    @ResourceGroupPreparer(name_prefix='cli_test_cannotdelete_lock_create_list_delete_resource_group')
+    def test_cannotdelete_lock_create_list_delete_resource_group(self, resource_group):
```
Comment (tjprescott) [inline src/command_modules/azure-cli-resource/azure/cli/command_modules/resource/tests/test_locks.py]: Rec: test_cannotdelete_resource_group_lock

File: src/command_modules/azure-cli-resource/azure/cli/command_modules/resource/tests/test_locks.py [hunk 3]
```diff
@@ -29,31 +29,60 @@ def test_lock_create_list_delete(self):
...
+        self._lock_operation_with_resource_group('CanNotDelete', resource_group)
 
-    def _lock_operation_with_resource_group(self, lock_type):
-        rg_name = self.create_random_name('cli.lock.rg', 75)
-        lock_name = self.create_random_name('cli-test-lock', 48)
+    @ResourceGroupPreparer(name_prefix='cli_test_readonly_lock_create_list_delete_resource')
+    def test_readonly_lock_create_list_delete_resource(self, resource_group):
```
Comment (tjprescott) [inline src/command_modules/azure-cli-resource/azure/cli/command_modules/resource/tests/test_locks.py]: Rec: test_readonly_resource_lock

File: src/command_modules/azure-cli-resource/azure/cli/command_modules/resource/tests/test_locks.py [hunk 4]
```diff
@@ -29,31 +29,60 @@ def test_lock_create_list_delete(self):
...
+        self._lock_operation_with_resource('ReadOnly', resource_group)
+
+    @ResourceGroupPreparer(name_prefix='cli_test_cannotdelete_lock_create_list_delete_resource')
+    def test_cannotdelete_lock_create_list_delete_resource(self, resource_group):
```
Comment (tjprescott) [inline src/command_modules/azure-cli-resource/azure/cli/command_modules/resource/tests/test_locks.py]: Rec: test_cannotdelete_resource_lock

Comment (williexu) [reply src/command_modules/azure-cli-resource/azure/cli/command_modules/resource/tests/test_locks.py]: Was considering doing so lol, will do

File: src/command_modules/azure-cli-resource/azure/cli/command_modules/resource/tests/test_locks.py [hunk 5]
```diff
@@ -29,31 +29,60 @@ def test_lock_create_list_delete(self):
             self.cmd('az lock delete -n {}'.format(lock_name))
             self._sleep_for_lock_operation()
 
-    def test_readonly_lock_create_list_delete_resource_group(self):
-        self._lock_operation_with_resource_group('ReadOnly')
+    @ResourceGroupPreparer(name_prefix='cli_test_readonly_resource_group_lock')
```
Comment (troydai) [inline src/command_modules/azure-cli-resource/azure/cli/command_modules/resource/tests/test_locks.py]: I don't think the prefix is necessary.

Comment (tjprescott) [reply src/command_modules/azure-cli-resource/azure/cli/command_modules/resource/tests/test_locks.py]: I recommend it :rofl:

**Comment by tjprescott** [comment]
Your PR is in violation of the sacred PEP8...

**Comment by williexu** [comment]
Duly noted

**Comment by troydai** [comment]
3cfe69a086c4574c607dfaaa988e26fa--iron-throne-spoiler

**Comment by williexu** [comment]
calm down folks

**Pull request merged**